File: trunk/PhanfareJavaAPI/src/com/phanfare/api/Session.java
package com.phanfare.api;

import org.json.me.JSONException;
import org.json.me.JSONObject;

public class Session {
	public String sessionCookie;
	public long userId;
	public boolean isPremium;
	public long friendGroupId;
	public long familyGroupId;
	public String websiteTitle;
	public String timelessHeader;
	public boolean isTimelessFirst;
	public PublicProfile profile;

	public static Session fromJson(JSONObject object) {
		Session session = new Session();
		try {
			session.sessionCookie = object.getString("cookie");
			session.userId = object.getLong("uid");
			session.isPremium = object.getBoolean("premium");
			session.friendGroupId = object.getLong("friend_group_id");
			session.familyGroupId = object.getLong("family_group_id");
			session.websiteTitle = object.getString("website_title");
			session.timelessHeader = object.getString("timeless_header");
			session.isTimelessFirst = object.getBoolean("timeless_first");
			session.profile = PublicProfile.fromJson(object.getJSONObject("profile"));
		} catch (JSONException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return null;
		}
		return session;
	}
}

File: Phanfare.ExternalAPI/Session.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace Phanfare.ExternalAPI
{
	public class Session
	{
		public string SessionCookie { get; set; }
		public long UserID { get; set; }
		public bool IsPremium { get; set; }
		public long FriendGroupID { get; set; }
		public long FamilyGroupID { get; set; }
		public long PublicGroupID { get; set; }
		public string WebsiteTitle { get; set; }
		public string TimelessHeader { get; set; }
		public bool IsTimelessFirst { get; set; }
		public PublicProfile Profile { get; set; }

		public static Session FromXML( XmlElement el )
		{
			Session session = new Session();
			session.SessionCookie = el.GetAttribute( "cookie" );
			session.UserID = el.GetAttributeInt64( "uid" );
			session.IsPremium = el.GetAttributeBoolean( "premium" );
			session.FriendGroupID = el.GetAttributeInt64( "friend_group_id" );
			session.FamilyGroupID = el.GetAttributeInt64( "family_group_id" );
			session.PublicGroupID = el.GetAttributeInt64( "public_group_id" );
			session.WebsiteTitle = el.GetAttribute( "website_title" );
			session.TimelessHeader = el.GetAttribute( "timeless_header" );
			session.IsTimelessFirst = el.GetAttributeBoolean( "timeless_first" );
			session.Profile = PublicProfile.FromXML( el[ "profile" ] );
			return session;
		}
	}
}
